pipeline/metadata/satellite.py

```diff
@@ -375,86 +356,191 @@ def _flat_rows_controls(key: Any, value: Row) -> Iterator[Tuple[Row, int]]:  # p
+
+def add_received_ip_tags(
+    rows: beam.pvalue.PCollection[Row],
+    tags: beam.pvalue.PCollection[Row]) -> beam.pvalue.PCollection[Row]:
+  """Add ip metadata info to received ip lists in rows
 
   Args:
-    row element like
+    rows: PCollection of dicts without metadata in the received ips like:
       {
-        "field": "value"
-        "received" [{
-            'ip': '1.2.3.4'
-        },{
+        'ip': '1.2.3.4'
+        'domain': 'ex.com'
+        'received' : [{
+            'ip': '4.5.6.7'
+          }, {
             'ip': '5.6.7.8'
-        }]
+          },
+        ]
       }
-
-  Returns:
-    Rows like
+    tags: PCollection of received ip metadata like:
       {
-        "field": "value"
-        "roundtrip_id" = "a"
-        "received" [{
-            'ip': '1.2.3.4'
-        }]
+        'ip': '4.5.6.7'
+        'asname': 'AMAZON-AES',
+        'asnum': 14618,
       }
       {
-        "field": "value"
-        "roundtrip_id" = "a"
-        "received" [{
-            'ip': '5.6.7.8'
-        }]
+        'ip': '5.6.7.8'
+        'asname': 'CLOUDFLARE',
+        'asnum': 13335,
       }
+
+  Returns a PCollection of rows with metadata tags added to received ips like
+    {
+      'ip': '1.2.3.4'
+      'domain': 'ex.com'
+      'received' : [{
+          'ip': '4.5.6.7'
+          'asname': 'AMAZON-AES',
+          'asnum': 14618,
+        }, {
+          'ip': '5.6.7.8'
+          'asname': 'CLOUDFLARE',
+          'asnum': 13335,
+        },
+      ]
+    }
   """
-  # This id is used to reconstruct the row structure when unflattening
-  dns_roundtrip['roundtrip_id'] = uuid.uuid4().hex
+  # PCollection[Row]
+  received_ip_tags = (
+      tags | 'filter out vp tags' >> beam.Filter(
+          lambda tag: not _is_vantage_point_tag(tag)).with_output_types(Row))
+
+  # PCollection[Tuple[DateIpKey, row]]
+  tags_keyed_by_ip_and_date = (
+      received_ip_tags | 'tags: key by ips and dates' >>
+      beam.Map(lambda tag: (make_date_ip_key(tag), tag)).with_output_types(
+          Tuple[DateIpKey, Row]))
 
-  all_received = dns_roundtrip['received']
+  # PCollection[Row]
+  rows_with_roundtrip_id = (
+      rows | 'add roundtrip_ids' >>
+      beam.Map(_set_random_roundtrip_id).with_output_types(Row))
+
+  # PCollection[row] received ip row
+  received_ips_with_roundtrip_ip_and_date = (
+      rows_with_roundtrip_id | 'get received ips' >> beam.FlatMap(
+          _get_received_ips_with_roundtrip_id_and_date).with_output_types(Row))
+
+  # PCollection[Tuple[DateIpKey, row]] received ip row
+  received_ips_keyed_by_ip_and_date = (
+      received_ips_with_roundtrip_ip_and_date |
+      'received ip: key by ips and dates' >>
+      beam.Map(lambda row: (make_date_ip_key(row), row)).with_output_types(
+          Tuple[DateIpKey, Row]))
 
-  if len(all_received) == 0:
-    yield dns_roundtrip
-    return
+  grouped_metadata_and_received_ips = (({
+      IP_METADATA_PCOLLECTION_NAME: tags_keyed_by_ip_and_date,
```

Comment:
Let's change the language here to match the language of merge_metadata_with_rows, and remove that extra param.

Reply:
done